src/main/java/com/cisco/trex/stateful/api/lowlevel/ASTFAssociationRule.java

```diff
@@ -26,13 +30,42 @@ public ASTFAssociationRule(String ipStart, String ipEnd, int port) {
     }
   }
 
+  /** @param port */
+  public ASTFAssociationRule(int port) {
+    this(null, null, port);
+  }
+
+  /**
+   * constructor
+   *
+   * @param ipStart
+   * @param ipEnd
+   * @param port
+   * @param l7List
+   */
+  public ASTFAssociationRule(String ipStart, String ipEnd, int port, List<Integer> l7List) {
+    this(null, null, port);
+    if (!l7List.isEmpty()) {
+      Map<String, List<Integer>> l7Map = new HashMap<>();
+      l7Map.put("offset", l7List);
+      fields.addProperty("l7_map", String.valueOf(l7Map));
```

Comment:
be sure the format is right?
String.valueOf will generate the ouput like: {key1=value1, key2=value2}

Reply:
done